scalyr_agent/builtin_monitors/kubernetes_monitor.py

```diff
@@ -3044,26 +3048,29 @@ def check_containers(self, run_state):
                                 limit_key="check-container-pod-info-%s" % cid,
                             )
 
+                        namespace = self.k8s_cache.namespace(pod_namespace, current_time=current_time, allow_expired=False)
                     # start the container if have a container that wasn't running
                     if cid not in self.containers:
                         self._logger.log(
                             scalyr_logging.DEBUG_LEVEL_1,
                             "Starting loggers for container '%s'" % info["name"],
                         )
                         starting[cid] = info
-                    elif cid in prev_digests:
+                    elif cid in prev_digests or cid in prev_digests_namespaces:
                         # container was running and it exists in the previous digest dict, so see if
                         # it has changed
-                        if pod and prev_digests[cid] != pod.digest:
+                        if pod and prev_digests[cid] != pod.digest or namespace and prev_digests_namespaces[cid] != namespace.digest:
```

Comment:
May want to add parenthesis around the whole `or` part of the if clause.

Reply:
Yes, that will make it more readable.